docs/cookbook/creation-of-custom-code-sniff.md

```diff
@@ -0,0 +1,120 @@
+    
+    }
+    
+    ```
+2. Make `register` method to return all [token constants](http://php.net/manual/en/tokens.php) you need to process.
+In this case we will use `T_NEW` constant so we will process all parts of code where objects are created.
+3. Enhance `process` method for code validation. We need to get full namespace path of the class and 
+check whether there exists its `Factory` and in that case whether the instantiation is not made outside the `Factory`.
+```php
+<?php
+
+declare(strict_types=1);
+
+namespace Shopsys\CodingStandards\Sniffs;
+
+use PHP_CodeSniffer\Files\File;
+use PHP_CodeSniffer\Sniffs\Sniff;
+use SlevomatCodingStandard\Helpers\ClassHelper;
+use Symplify\TokenRunner\Analyzer\SnifferAnalyzer\Naming;
+
+final class ObjectIsCreatedByFactorySniff implements Sniff
+{
+    /**
+     * @var Naming
+     */
+    private $naming;
+
+    public function __construct(Naming $naming)
+    {
+        $this->naming = $naming;
+    }
+
+    /**
+     * @return int[]
+     */
+    public function register(): array
+    {
+        return [T_NEW];
+    }
+
+    /**
+     * @param File $file
+     * @param int $position
+     */
+    public function process(File $file, $position): void
+    {
+        // get full name of the file class
+        $fileClassName = ClassHelper::getFullyQualifiedName($file, $file->findNext(T_CLASS, 2));
+
+        // get full name of the class that is instantiated inside of some method of the file class
+        $className = '\\' . $this->naming->getClassName($file, $file->findNext(T_STRING, $position));
+        $factoryName = $className . 'Factory';
+
+        // if instantiated class is instantiated inside 
+        // if instantiated class doesn't have factory in the same namespace path then code is valid
+        if ($factoryName === $fileClassName || !class_exists($factoryName)) {
+            return;
+        }
+
+        $file->addError(
+            sprintf('For creation of "%s" class use its factory "%s"', $className, $factoryName),
+            $position,
+            self::class
+        );
+    }
+}
+```
+
+## Integration of sniff into coding standards
+For sniff to be sniffing we need to register sniff class as a service
+in [easy coding standards](https://github.com/Symplify/EasyCodingStandard) config. In this case we have our own
+[coding standards](https://github.com/shopsys/coding-standards/) config located in file `shopsys-coding-standard.yml`,
+so in the services section we add a line with `Shopsys\CodingStandards\Sniffs\ObjectIsCreatedByFactorySniff: ~`.
+```yml
+services:
+    Shopsys\CodingStandards\Sniffs\ObjectIsCreatedByFactorySniff: ~
+```
+For excluding some files from checking of this sniff rule there is a way by defining row in skip subsection of parameters section.
+```yml
+parameters:
+    skip:
+        Shopsys\CodingStandards\Sniffs\ObjectIsCreatedByFactorySniff:
+            - '*/tests/*'
+```
+
+## Analyze helpers
```

Comment:
it was not my point :laughing: I just wanted you to use the term "Analysis helpers" instead of the original "Analyze helpers" but never mind :slightly_smiling_face: